scipy/interpolate/_rbfinterp.py

```diff
@@ -0,0 +1,538 @@
+        data_shape = d.shape[1:]
+        data_type = d.dtype
+        d = d.reshape((ny, -1))
+        d = d.view(float)
+        powers = _monomial_powers(ndim, degree)
+        tree = KDTree(y)
+
+        self.y = y
+        self.d = d
+        self.k = k
+        self.smoothing = smoothing
+        self.kernel = kernel
+        self.epsilon = epsilon
+        self.powers = powers
+        self.tree = tree
+        self.data_shape = data_shape
+        self.data_type = data_type
+
+    def __call__(self, x):
+        """
+        Evaluates the interpolant at `x`
+
+        Parameters
+        ----------
+        x : (Q, N) array_like
+            Interpolation point coordinates
+
+        Returns
+        -------
+        (Q, ...) ndarray
+            Values of the interpolant at `x`
+
+        """
+        x = np.asarray(x, dtype=float, order='C')
+        if x.ndim != 2:
+            raise ValueError('`x` must be a 2-dimensional array')
+
+        nx, ndim = x.shape
+        if ndim != self.y.shape[1]:
+            raise ValueError(
+                'Expected the second axis of `x` to have length %d' %
+                self.y.shape[1]
+                )
+
+        # get the indices of the k nearest observation points to each
+        # interpolation point
+        _, yindices = self.tree.query(x, self.k)
+        if self.k == 1:
+            # KDTree squeezes the output when k=1
+            yindices = yindices[:, None]
+
+        # multiple interpolation points may have the same neighborhood of
+        # observation points. Make the neighborhoods unique so that we only
+        # compute the interpolation coefficients once for each neighborhood
+        yindices = np.sort(yindices, axis=1)
+        yindices, inv = np.unique(yindices, return_inverse=True, axis=0)
+        # `inv` tells us which neighborhood will be used by each interpolation
+        # point. Now we find which interpolation points will be using each
+        # neighborhood
+        xindices = [[] for _ in range(len(yindices))]
+        for i, j in enumerate(inv):
+            xindices[j].append(i)
+
+        out = np.empty((nx, self.d.shape[1]), dtype=float)
+        for xidx, yidx in zip(xindices, yindices):
+            # `yidx` are the indices of the observations in this neighborhood.
+            # `xidx` are the indices of the interpolation points that are using
+            # this neighborhood
+            xnbr = x[xidx]
+            ynbr = self.y[yidx]
+            dnbr = self.d[yidx]
+            snbr = self.smoothing[yidx]
+            shift, scale, coeffs = _build_and_solve_system(
+                ynbr, dnbr, snbr, self.kernel, self.epsilon, self.powers,
+                )
+
+            out[xidx] = _evaluate(
+                xnbr, ynbr, self.kernel, self.epsilon, self.powers, shift,
+                scale, coeffs
+                )
+
+        out = out.view(self.data_type)
+        out = out.reshape((nx,) + self.data_shape)
+        return out
```

Comment:
End file with newline.
```suggestion
        return out

```